### PR TITLE
8262862: Harden tests sun/security/x509/URICertStore/ExtensionsWithLDAP.java and krb5/canonicalize/Test.java

### PR DESCRIPTION
**Pull request by fguallini**
Certain JVM tools may access and initialise InetAddress class and its static fields during start up resulting in a NameService implementation being already set to the default **PlatformNameService**, causing intermittent failures in some tests that expect the use of **HostsFileNameService** instead by setting the system property jdk.net.hosts.file in the test main body, which has no effect in this scenario

By setting the property **jdk.net.hosts.file** in the @run tag, there is certainty that it will be properly configured at the time that a [NameService](https://github.com/openjdk/jdk/blob/382e38dd246596ec94a1f1ce0e0f9e87f53366c7/src/java.base/share/classes/java/net/InetAddress.java#L1155) is created

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8262862](https://bugs.openjdk.java.net/browse/JDK-8262862): Harden tests sun/security/x509/URICertStore/ExtensionsWithLDAP.java and krb5/canonicalize/Test.java

### Reviewers
 * [Aleksei Efimov](https://openjdk.java.net/census#aefimov) (@AlekseiEfimov - Committer)
 * [Michael McMahon](https://openjdk.java.net/census#michaelm) (@Michael-Mc-Mahon - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2809/head:pull/2809`
`$ git checkout pull/2809`

### CODE REVIEW

File: test/jdk/sun/security/krb5/canonicalize/Test.java [hunk 1]
```diff
@@ -25,18 +25,15 @@
...
-import java.net.InetAddress;
-import java.net.UnknownHostException;
 import sun.security.krb5.PrincipalName;
 
 public class Test {
     public static void main(String[] args) throws Exception {
         // This config file is generated using Kerberos.app on a Mac
-        String hostsFileName = System.getProperty("test.src", ".") + "/TestHosts";
-        System.setProperty("jdk.net.hosts.file", hostsFileName);
         System.setProperty("java.security.krb5.realm", "THIS.REALM");
```
Comment (AlekseiEfimov) [inline test/jdk/sun/security/krb5/canonicalize/Test.java:37]: Maybe, you can move setting of `java.security.krb5.realm` and `java.security.krb5.kdc` properties to the `jtreg` header to have consistent approach of setting properties needed for a test run.

File: test/jdk/sun/security/x509/URICertStore/ExtensionsWithLDAP.java [hunk 2]
```diff
@@ -136,11 +135,6 @@ public static void main(String[] args) throws Exception {
         System.setProperty("com.sun.security.enableCRLDP", "true");
```
Comment (AlekseiEfimov) [inline test/jdk/sun/security/x509/URICertStore/ExtensionsWithLDAP.java:135]: Same suggestion for `com.sun.security.enableCRLDP` and `com.sun.security.enableAIAcaIssuers` properties.

**Review by AlekseiEfimov — APPROVED** [review]
Hi Fernando,

Changes look good to me.
Have one minor suggestion to also move other system properties to jtreg header:

File: test/jdk/sun/security/krb5/canonicalize/Test.java (hunk 1 above)
Comment (fguallini) [reply test/jdk/sun/security/krb5/canonicalize/Test.java]: Agree. Updated in latest commit, thanks

**Review by AlekseiEfimov — APPROVED** [review]
The latest version looks good to me. Thanks.

**Review by Michael-Mc-Mahon — APPROVED** [review]
LGTM

**Pull request closed without merging**